stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -662,7 +662,7 @@ macro artifact_str(name, platform=nothing, artifacts_toml_path=nothing)
     local artifacts_toml = if artifacts_toml_path === nothing || artifacts_toml_path == :(nothing)
         find_artifacts_toml(srcfile)
     else
-        eval(artifacts_toml_path)
+        Core.eval(__module__, artifacts_toml_path)
```

Comment:
Instead of `eval` use `getfield`? But that may just be personal taste.

Reply:
How would you use `getfield`?  In this case, `artifacts_toml_path` could be an expression, rather than a quoted string.

Reply:
I would prefer not using a runtime/order dependent lookup here. `eval` inside a macro is for me a code-smell. Either the data is a literal and can be relied upon, or it is a runtime value and we should defer it's evaluation to inside the `quote` block.